### ISSUE TITLE
bug: E2EPage setContent removes all script and style tags from the page

### ISSUE THREAD

**Issue by Trendy** [ionic-team/stencil#3229]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.13

### Current Behavior

Sometimes there may be external dependencies that many tests will need, so adding a script or stylesheet to the E2EPage could be defined in a `beforeEach/beforeAll`. However, if you are using `E2EPage.setContent` in an individual test, it will remove all previously added script and styles, leading to some confusing behavior. 

### Expected Behavior

Calling E2EPage.setContent will not remove previous scripts and styles.

### Steps to Reproduce

1. Create an end-to-end test for a component.
2. In the test, add a `beforeEach` that creates the page, and calls `page.addScriptTag` to add a dependency (i.e. bootstrap, shoelace, etc).
3. In the test, add a `describe` and `it`, call `page.setContent`.
4. Launch the test in browser mode with debugging enabled, inspect the `<head>`.

### Code Reproduction URL

https://github.com/Trendy/stencil-set-content-bug

**Comment by rwaskiewicz** [comment]
👋 

I've confirmed that this is a bug and have labeled it so that it gets ingested into our internal issue tracker & backlog. Thanks for the great reproduction case, it's very much appreciated!!

**Comment by Trendy** [comment]
No problem! 

I took a stab at fixing it last night as well, but I couldn't get the build to work linked into my project locally. I'll reach out to the Slack for some help with that later on today.

If anyone else is having issues with this, I was able to get around it by adding the script tags _after_ the `setContent` call. The dependency was another collection of UI components though, so your milage may vary if your component depends on external libraries for rendering or connectedCallbacks.

**Comment by Trendy** [comment]
**Update:I** revisited this issue, and came up with a solution. The pull request is failing automated tests, currently. I'll attempt to fix those in the future.

**Comment by christian-bromann** [comment]
The idea of `E2EPage.setContent` I believe is to guarantee that the user renders a clean page without any side effects due to scripts that kept around from previous tests.

> Sometimes there may be external dependencies that many tests will need

Can you clarify what types of scripts or stylesheets you like to keep around?

I think a better approach would be to allow fixtures (as html files, scripts or stylesheets that you can pass into the `E2EPage.setContent` method that allow you to construct your test environment implicitly. What do you think?

**Comment by Trendy** [comment]
I can see the logic there. In my case I was recreating the page for each test. What I ended up doing to get around it was setting the content, then adding the script tags/stylesheets I required from another library. 

I haven't looked at the code for `setContent` in a while, but I think your idea would be a viable solution and make things cleaner for my code base! 

I wouldn't mind taking a stab at implementing this if you all are too busy, I would want some input on the API though! 👍
